docs: update Thread CoAP client MED/SED info (#2786)
nrfconnect/sdk-nrf: samples/openthread/coap_client/README.rst

Testing Minimal Thread Device
-----------------------------

After building the MTD variant of this sample and programming it, the device starts in the MED mode with **LED 3** on.

> This means that the radio is enabled, but the serial console is disabled to save power.

Reviewer: ```suggestion
This means that the radio is enabled when idle and the serial console is operating.
```

You can switch to the SED mode at any moment during the standard testing procedure.

To toggle SED, press **Button 3** on the client node.
**LED 3** turns off to indicate the switch to the SED mode.

> At this point, the radio is disabled and power consumption decreases.

Reviewer: ```suggestion
At this point, the radio is disabled when idle and the serial console is not working to decrease the power consumption.
```

Pressing **Button 3** again will switch the mode back to MED.

> Switching between SED and MED modes does not affect the standard testing procedure.

Reviewer: Not sure if we should expand this part, leaving it to your judgement.
```suggestion
Switching between SED and MED modes does not affect the standard testing procedure, except for the fact that LEDs and terminal logs are not available in the SED mode.
```

Reviewer: My bad, LEDs work just fine in the SED mode. Only the LED3 is turned off.
